tutorials/terraform/data-proc-data-exchange-with-mch.tf

```
@@ -0,0 +1,281 @@
+    from_port         = 0
+    to_port           = 65535
+    predefined_target = "self_security_group"
+  }
+
+  egress {
+    description    = "Allow connections to the HTTPS port."
+    protocol       = "TCP"
+    port           = 443
+    v4_cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "yandex_iam_service_account" "dataproc" {
+  description = "Service account to manage the Data Proc cluster."
+  name        = local.dp_account
+}
+
+# Roles to create Data Proc cluster.
+resource "yandex_resourcemanager_folder_iam_binding" "dataproc-agent" {
+  folder_id = local.folder_id
+  role      = "dataproc.agent"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+resource "yandex_resourcemanager_folder_iam_binding" "dataproc-provisioner" {
+  folder_id = local.folder_id
+  role      = "dataproc.provisioner"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+resource "yandex_resourcemanager_folder_iam_binding" "monitoring-viewer" {
+  folder_id = local.folder_id
+  role      = "monitoring.viewer"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+# Role to create Object Storage Bucket
+resource "yandex_resourcemanager_folder_iam_binding" "bucket-creator" {
+  folder_id = local.folder_id
+  role      = "storage.editor"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+resource "yandex_mdb_clickhouse_cluster" "clickhouse-cluster" {
+  description        = "Managed Service for ClickHouse cluster"
+  name               = "clickhouse-cluster"
+  environment        = "PRODUCTION"
+  network_id         = local.network_id
+  security_group_ids = [yandex_vpc_security_group.clickhouse-and-vm-security-group.id]
+
+  clickhouse {
+    resources {
+      resource_preset_id = "s2.micro" # 2 vCPU, 8 GB RAM
+      disk_type_id       = "network-ssd"
+      disk_size          = 10 # GB
+    }
+  }
+
+  host {
+    type             = "CLICKHOUSE"
+    zone             = local.zone_id
+    subnet_id        = local.subnet_id
+    assign_public_ip = true # Required for connection from the Internet
+  }
+
+  database {
+    name = "db1"
+  }
+
+  user {
+    name     = "user1"
+    password = local.ch_password
+    permission {
+      database_name = "db1"
+    }
+  }
+}
+
+# VM in Yandex Compute Cloud
+resource "yandex_compute_instance" "vm-1" {
+  name        = "linux-vm"
+  platform_id = "standard-v3" # Intel Ice Lake
+  zone        = local.zone_id
+
+  resources {
+    cores  = 2
+    memory = 4 # GB
+  }
+
+  boot_disk {
+    initialize_params {
+      image_id = local.vm_image_id
+    }
+  }
+
+  network_interface {
+    subnet_id          = local.subnet_id
+    security_group_ids = [yandex_vpc_security_group.clickhouse-and-vm-security-group.id]
+    nat                = true # Required for connection from the Internet
+  }
+
+  metadata = {
+    ssh-keys = "${local.vm_username}:${file(${local.vm_ssh_key})}"
```

Сложная конструкция. Пожалуйста, проверь, можно ли сократить до такого?

```suggestion
    ssh-keys = "${local.vm_username}:file(${local.vm_ssh_key})"
```

---

Да, так как ты предложил и нужно писать. Иначе `terraform fmt` ругается.